doc/ref/plotting_options/styling.ipynb

```diff
@@ -0,0 +1,742 @@
+    "- No extra setup is required\n",
+    "- Colormaps work consistently across all visualizations\n",
+    "- Defaults like `glasbey_category10` and `kbc_r` come from `Colorcet`\n",
+    "\n",
+    "This makes it a reliable and visually accessible choice for both scientific and presentation-quality plots.\n",
+    ":::"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1c34a309-d7c0-423e-adc7-9743f3d7a331",
+   "metadata": {},
+   "source": [
+    "(option-colormap)=\n",
+    "## `colormap`\n",
+    "\n",
+    "Alias for [`cmap`](option-cmap) above."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3290b419-10a3-4d2b-ac99-0e50c644fbcc",
+   "metadata": {},
+   "source": [
+    "(option-color_key)=\n",
+    "## `color_key`\n",
+    "\n",
+    "The `color_key` option defines a categorical color mapping, primarily used with `datashade=True`. It maps distinct categories in your dataset to specific colors.\n",
+    "\n",
+    "It accepts:\n",
+    "- A list of colors (matched to unique categories in order)\n",
+    "- A dictionary mapping category values to colors\n",
+    "\n",
+    "If not provided, hvPlot uses a default categorical colormap. Only one of `cmap`, `colormap`, or `color_key` should be used at a time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c7c13149-05de-4fcd-96eb-6bd6e84dea06",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import hvplot.pandas  # noqa\n",
+    "import hvsampledata\n",
+    "\n",
+    "df = hvsampledata.earthquakes(\"pandas\")\n",
+    "\n",
+    "# Define explicit color mapping for depth_class\n",
+    "depth_colors = {\n",
+    "    'Shallow': '#FFC300',\n",
+    "    'Intermediate': '#C70039',\n",
+    "    'Deep': '#581845'\n",
+    "}\n",
+    "\n",
+    "df.hvplot.scatter(\n",
+    "    x='lon',\n",
+    "    y='lat',\n",
+    "    c='depth_class',\n",
+    "    color_key=depth_colors,\n",
+    "    title=\"Earthquakes with Custom Color Key\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e0156152-63bc-4f26-9b0c-6c2e5c2a0ab5",
+   "metadata": {},
+   "source": [
+    "::: {note}\n",
+    "\n",
+    "**Choosing Between `cmap` and `color_key`**\n",
+    "\n",
+    "When creating datashaded plots (`datashade=True` or `rasterize=True`), you should choose between `cmap` and `color_key` based on the type of data you're mapping to color:\n",
+    "\n",
+    "- Use **`cmap`** for **continuous numerical values** (e.g., depth, magnitude). It maps a range of values to a color gradient.\n",
+    "- Use **`color_key`** for **categorical values** (e.g., mag_class, depth_class). It maps each category explicitly to a color.\n",
+    "\n",
+    "For example:\n",
+    "- `cmap='viridis'` is great for smooth gradients like 'magnitude' or 'depth'.\n",
+    "- `color_key={'Shallow': 'blue', 'Deep': 'green'}` is best for labeled classes like 'depth_class'.\n",
```

Comment:
So for a datashaded plot we cannot use `cmap={'Shallow': 'blue', 'Deep': 'green'}`?

Reply:
It works for categorical columns because internally `cmap` becomes `color_key`:
```python
if 'cmap' in self._style_opts and self.datashade:
    ...
    if isinstance(cmap, dict):
        opts['color_key'] = cmap
    else:
        opts['cmap'] = process_cmap(cmap, levels, categorical=categorical)
```
https://github.com/holoviz/hvplot/blob/main/hvplot/converter.py#L1977-L1983

The difference is when you want to map colors for use in continuous columns. I will update the explanation and examples used.

Reply:
Ok thanks. I'm trying to understand why `color_key` is needed, and if not, I'd be really happy to deprecate it (having `colormap` and `cmap` is already enough...). I am trying to avoid documenting too much a feature I'd rather hide if it doesn't bring much to the table.